### DISCUSSION TITLE
VL1L2 and VCNT lines missing

### DISCUSSION THREAD

**Discussion by CPKalb** [dtcenter/METplus#1846] [Configuration] [answered]
Hi:

I'm struggling to get VL1L2 and VCNT lines out from point_stat.  I have it working correctly on another dataset, but I'm getting no output on another.  Wind speed is being derived.  Relevant parts of my configuration file are located here:
POINT_STAT_OUTPUT_FLAG_VL1L2 = STAT
POINT_STAT_OUTPUT_FLAG_VCNT = STAT

FCST_POINT_STAT_VAR4_NAME = WIND
FCST_POINT_STAT_VAR4_LEVELS = Z10
FCST_POINT_STAT_VAR4_THRESH = ge10, ge20

FCST_POINT_STAT_VAR5_NAME = UGRD
FCST_POINT_STAT_VAR5_LEVELS = Z10
FCST_POINT_STAT_VAR5_OPTIONS = is_u_wind = TRUE;

FCST_POINT_STAT_VAR6_NAME = VGRD
FCST_POINT_STAT_VAR6_LEVELS = Z10
FCST_POINT_STAT_VAR6_OPTIONS = is_v_wind = TRUE;

OBS_POINT_STAT_VAR4_NAME = wspeed
OBS_POINT_STAT_VAR4_LEVELS = Z10
OBS_POINT_STAT_VAR4_THRESH = ge10, ge20

OBS_POINT_STAT_VAR5_NAME = uwind
OBS_POINT_STAT_VAR5_LEVELS = Z10
OBS_POINT_STAT_VAR5_OPTONS = is_u_wind = TRUE;

OBS_POINT_STAT_VAR6_NAME = vwind
OBS_POINT_STAT_VAR6_LEVELS = Z10
OBS_POINT_STAT_VAR6_OPTONS = is_v_wind = TRUE;

I've attached a log file, with the verbosity set to 5.  But, I can't find any mention of the VL1L2 or VCNT lines in there
[metplus.log.20221004195340.txt](https://github.com/dtcenter/METplus/files/9710201/metplus.log.20221004195340.txt)

Any idea what I am missing?

Thanks,
Tina

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @CPKalb, I see you're having trouble getting VL1L2 and VCNT output when you run Point-Stat.

I see that you're identifying the data as being the U and V component of winds in your METplus config file using:
```
FCST_POINT_STAT_VAR5_OPTIONS = is_u_wind = TRUE;
FCST_POINT_STAT_VAR6_OPTIONS = is_v_wind = TRUE;
OBS_POINT_STAT_VAR5_OPTONS = is_u_wind = TRUE;
OBS_POINT_STAT_VAR6_OPTONS = is_v_wind = TRUE;
```
I see the impact of the `FCST` settings in the log file you provided:
```
10/04 19:53:40.657Z metplus INFO: METPLUS_FCST_FIELD=field = [{ name="TMP"; level="Z2"; cat_thresh=[ le20,le15,ge30,ge35,ge40,ge45 ]; convert(x) = K_to_C(x); },{ name="DPT"; level="Z2"; cat_thresh=[ le10,ge15,ge20,ge25 ]; convert(x) = K_to_C(x); },{ name="RH"; level="Z2"; cat_thresh=[ ge50,ge60,ge70 ]; },{ name="WIND"; level="Z10"; cat_thresh=[ ge10,ge20 ]; },{ name="UGRD"; level="Z10"; is_u_wind = TRUE; },{ name="VGRD"; level="Z10"; is_v_wind = TRUE; }];
```
However I **DO NOT** see the impact of the `OBS` settings in that log file:
```
10/04 19:53:40.658Z metplus INFO: METPLUS_OBS_FIELD=field = [{ name="Dry_Temp"; level="Z2"; cat_thresh=[ le20,le15,ge30,ge35,ge40,ge45 ]; },{ name="Dewpoint"; level="Z2"; cat_thresh=[ le10,ge15,ge20,ge25 ]; },{ name="RH"; level="Z2"; cat_thresh=[ ge50,ge60,ge70 ]; },{ name="wspeed"; level="Z10"; cat_thresh=[ ge10,ge20 ]; },{ name="uwind"; level="Z10"; },{ name="vwind"; level="Z10"; }];
```
Note that `is_u_wind` and `is_v_wind` are not present for the OBS settings.

Please confirm that your METplus conf file actually does contain those settings.

If so, @georgemccabe can you please advise on why the `OBS_POINT_STAT_VAR<n>_OPTIONS` aren't making their way into the actual configuration that is run?

**Reply by CPKalb** [reply]
Ah, I see the problem.  They "looked" like they were present in the obs data.  However, the word OPTIONS spelled wrong.  So, that's why they are not making it in to the options for the point_stat run.  Sorry about not catching that typo.

Tina
